test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,485 @@
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 *10  = 300 bytes
+        writeEvents(10, writer);
+        writer.flush();
+
+        //Requested next stream cut at a distance of 170 bytes, and getting the next approx offset as a response.
+        StreamCut streamCut1 = batchClient.getNextStreamCut(streamCut0, 170L);
+        long streamCut1Position = streamCut1.asImpl().getPositions().get(list.get(0)).longValue();
+        log.info("Next stream cut1 {} streamCut1 position {}", streamCut1, streamCut1Position);
+        assertTrue(180 <= streamCut1Position);
+
+        @Cleanup
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(streamScope, controllerURI);
+        ReaderGroupConfig readerGroupConfig1 = getReaderGroupConfig(streamCut0, streamCut1, stream);
+        groupManager.createReaderGroup(readerGroupName, readerGroupConfig1);
+
+        @Cleanup
+        ReaderGroup readerGroup = groupManager.getReaderGroup(readerGroupName);
+
+        // Reading the events between the two streamCut, startStreamCut = streamCut0 and endStreamCut = streamCut1
+        @Cleanup
+        EventStreamReader<String> reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+
+        int readCount1 = readEvent(reader0, streamCut1Position / 30);
+        reader0.close();
+        assertEquals(readCount1, streamCut1Position / 30);
+
+        StreamCut streamCut2 = batchClient.getNextStreamCut(streamCut1, 80L);
+        long streamCut2Position = streamCut2.asImpl().getPositions().get(list.get(0)).longValue();
+        assertTrue(270 <= streamCut2Position);
+        long distanceBetweenTheStreamCut2 = streamCut2Position - streamCut1Position;
+        log.info("Next stream cut2 {} stream cut position {} distance between the stream cut {}", streamCut2, streamCut2Position, distanceBetweenTheStreamCut2);
+
+        ReaderGroupConfig readerGroupConfig2 = getReaderGroupConfig(streamCut1, streamCut2, stream);
+        readerGroup.resetReaderGroup(readerGroupConfig2);
+
+        // Reading the events between the two streamCut, startStreamCut = streamCut1 and endStreamCut = streamCut2
+        reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+
+        int readCount2 = readEvent(reader0, distanceBetweenTheStreamCut2 / 30);
+        reader0.close();
+        assertEquals(readCount2, distanceBetweenTheStreamCut2 / 30);
+
+        long approxDistanceToNextOffset = 350L;
+        StreamCut streamCut3 = batchClient.getNextStreamCut(streamCut2, approxDistanceToNextOffset);
+        long streamCut3Position = streamCut3.asImpl().getPositions().get(list.get(0)).longValue();
+        assertTrue(300 <= streamCut3Position);
```

Comment:
Since we have written only 10events, can we make this assertion as equals to 300 as this would be the tail offset for the segment.

Reply:
made the change